addons/sale_margin/models/sale_order.py

```diff
@@ -2,11 +2,18 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import api, fields, models
+from odoo.tools.sql import column_exists
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    def _auto_init(self):
```

Comment:
Now that i think of it, it should be done in a `pre_init_hook` instead (like it's done in other modules).
See https://github.com/odoo/odoo/pull/57850/commits/17f17884f8acf9981d6be73933181029aa30601a

Reply:
Okay, my bad, thought it was the standard solution, but both seem to be as good :D (we should harmonize that someday).
Thought `_auto_init` was called more frequently than the `pre_init_hook` and that it would be a bigger overhead, but doesn't seem to be the case ^^'.

Also, did you try using the `create_column` tooling method to create the column instead of raw sql (might be cleaner) ?

Reply:
@Feyensv I didn't use it because it's not possible to use DEFAULT

Reply:
But do you even need the default?

Reply:
@Feyensv the point of this PR is to make things faster so quoting from Postgres documentation the following: 
`From PostgreSQL 11, adding a column with a constant default value no longer means that each row of the table needs to be updated when the ALTER TABLE statement is executed. Instead, the default value will be returned the next time the row is accessed and applied when the table is rewritten, making the ALTER TABLE very fast even on large tables.`
So yep, I need that default thingy!

Reply:
You need the field to not be recomputed by the ORM on all existing records, that's the fix of your ticket AFAIK.
But why do you **need** the DEFAULT for your fix?  It's a reasonable 'hack' to create the column before the ORM does, since the result is the same, but it shouldn't have a different definition, unless really necessary.
Keep things as standard as possible, please.

Reply:
For reference: like you mentioned this fix is for 13.0 it would make sense to not use `default` as the use of pg11 is not guaranteed